torch/fx/experimental/sym_node.py

```diff
@@ -171,6 +178,28 @@ def is_float(self):
     def is_bool(self):
         return self.pytype is bool
 
+    def is_singleton(self):
+        return self._singleton_dummy is not None or (
+            # In case singleton SymNode is not fully initialized.
+            self.hint is not None
+            and isinstance(self.hint, SymInt)
+            and self.hint.node.is_singleton()
+        )
```

Comment:
this is nice; does it mean we can get rid of the `is_singleton()` definition in `torch/fx/experimental/symbolic_shapes.py`?

Reply:
Yup, thanks for the reminder :)

Reply:
Hmm unless we want a method on SymInt, going to keep the method for now, but at least now its going to be much simpler.